### PR TITLE
Add Hex4B PCB

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

Add Hex4B PCB, Atmega32a.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — COMMENTED** [review]
Looks like you need to rebase to fix the lib/chibios changes

**Review by drashna — APPROVED** [review]

File: keyboards/mechlovin/hex4b/rules.mk [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# MCU name
+MCU = atmega32a
+
+# Bootloader selection
+BOOTLOADER = USBasp
+
+F_CPU ?= 16000000
```
Comment (fauxpark) [inline keyboards/mechlovin/hex4b/rules.mk:7]:
```suggestion
# Processor frequency
F_CPU = 16000000
```

File: keyboards/mechlovin/hex4b/rules.mk [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
```
Comment (fauxpark) [inline keyboards/mechlovin/hex4b/rules.mk:21]:
```suggestion
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
```

File: keyboards/mechlovin/hex4b/readme.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# hex4b
+
+![hex4b](https://i.imgur.com/lArfGgH.jpeg)
```
Comment (fauxpark) [inline keyboards/mechlovin/hex4b/readme.md:3]:
```suggestion
![hex4b](https://i.imgur.com/lArfGgHh.jpeg)
```

File: keyboards/mechlovin/hex4b/readme.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+* Hardware Supported: Hex.4B PCB, Atmega32A 
+* Hardware Availability: [Hex Keyboard](https://hexkeyboards.com/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make mechlovin/hex_kb/hex4b:default
+
+Flashing example for this keyboard:
+
+    make mechlovin/hex_kb/hex4b:default:flash
```
Comment (fauxpark) [inline keyboards/mechlovin/hex4b/readme.md:17]:
```suggestion
    make mechlovin/hex4b:default

Flashing example for this keyboard:

    make mechlovin/hex4b:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**